pkg/k8s/pod.go

```diff
@@ -619,9 +620,11 @@ func (c *Client) preparePodSpec(spec PodConfig, init bool) v1.PodSpec {
 
 	// Prepare sidecar containers and append to the pod spec
 	for _, sidecarConfig := range spec.SidecarConfigs {
+		sidecarInitContainer := c.prepareInitContainers(sidecarConfig, true)
```

Comment:
[Q] how many init containers a pod can have? I thought it could be only one.

Reply:
There is no limit to the init container.
For example, if you have a main container and a sidecar (eg. otel-agent), then we now have 2 init containers.